web-components/src/components/robotoff-nutrients/robotoff-nutrients-table.ts

```diff
@@ -309,39 +322,93 @@ export class RobotoffNutrientsTable extends LitElement {
     )
   }
 
+  isUnitInput(key: string) {
+    return key.startsWith(NUTRIENT_UNIT_NAME_PREFIX)
+  }
+
   /**
-   * Handle the form submission.
-   * It will emit a custom submit event to submit the form data well formatted.
-   * It only sent the data for the column that was submitted.
-   * @param event
+   * Validate the input value.
    */
-  onSubmit(event: SubmitEvent) {
-    event.preventDefault()
-    event.stopPropagation()
-    const column = event.submitter?.getAttribute("data-key") as InsightAnnotationType
-    if (!column) {
-      console.error("No column found in submitter")
-      return
-    }
+  validateInputValue(value: string | null): {
+    error?: string
+    value?: string
+  } {
+    const valueCleaned = value?.replace(" ", "").replace(",", ".") ?? ""
 
-    const formData = new FormData(event.target as HTMLFormElement)
+    if (valueCleaned === "") {
+      return {
+        value: valueCleaned,
+      }
+    }
+    // Check if the value is a number and a multiple of 0.01. Replace < by nothing to allow to enter a value like <1
+    const numberValue = Number(valueCleaned.replace(/^</gm, ""))
+    if (isNaN(numberValue)) {
+      return {
+        error: msg("Error: Invalid value."),
+      }
+    }
```

Comment:
we should also support the "-" value which means: not present on packaging.

Reply:
only "-"

Reply:
removing the row is not really the same thing. Knowing that an important nutrient is not specified on the product label is an important information. (for example on fibers)